### DISCUSSION TITLE
Font sample BASH scripting

### DISCUSSION THREAD

**Discussion by greytomorrow** [ImageMagick/ImageMagick#2020] [General] [answered]
Hello, everybody.

I'm trying to write a stupid-simple BASH script to create a font sample image, but I'm getting errors on the path entries and fontnames with spaces.

```
#! /bin/bash

read -p "Drag font here: " FONT
echo $FONT

convert -size 800x -gravity Center -background white -fill black \
-font $FONT -pointsize 72 \
caption:@/00_alphabet.txt $FONT.png
```
I had a look at the documentation but I couldn't find anything regarding special characters in paths or filenames (but I could have missed it). Any suggestions? Thanks in advance.

**Comment by fmw42** [comment]
What is an example of FONT? Is it just a name or a path to a font file such as library/fonts/arial.ttf?

Is your text file "@/00_alphabet.txt" really at the root of your system? If not provide the proper path and not just /

**Comment by greytomorrow** [comment]
Thanks for the quick reply! 

> Is it just a name or a path to a font file such as library/fonts/arial.ttf?
Yes, it could be. But it coud also be any font on the hard drive, such as...

`/Users/grey/Archives/***FONTS/**pape/American Wood Type/AWTBill-StarkConcaveTuscanCond.ttf`

...the problem here being ImageMagick "chokes" at Wood, thinking it's the output file name. Fonts in directories with no spaces, or fonts with no spaces in the filename are no problem.

> Is your text file "@/00_alphabet.txt" really at the root of your system?

Actually it's not. It's in another folder, but that's really not here nor there; it doesn't matter where it resides, ImageMagick reads from it as expected. Here it is if you're interested...

[00_alphabet.txt](https://github.com/ImageMagick/ImageMagick/files/4630298/00_alphabet.txt)

**Reply by fmw42** [reply]
Put your path to the font file in double quotes. The chocking is due to the space in the folder "American Wood Type". So 

"/Users/grey/Archives/***FONTS/**pape/American Wood Type/AWTBill-StarkConcaveTuscanCond.ttf"

**Comment by greytomorrow** [comment]
If you would be so kind, how would I do that?

**Reply by fmw42** [reply]
Try

```
convert -size 800x -gravity Center -background white -fill black \
-font "$FONT" -pointsize 72 \
caption:@/00_alphabet.txt $FONT.png
```

**Comment by greytomorrow** [comment]
Tried that already (as well as '$FONT'). Thanks anyway.

**Reply by fmw42** [reply]
Single quotes will read $ literally and not as a variable.

If you do `echo "$FONT"` with the quotes, does it show the correct path information?

Alternately, rename that folder to remove the spaces.

**Comment by greytomorrow** [comment, marked as answer]
As much I as sincerely appreciate your help on this, I have the feeling we're not on the same page. If you try my script, after making any adjustments needed to run on your machine, you'll see the problem. Renaming the folder would defeat the purpose of the script. And, yes, `echo "$FONT"` shows the correct path.

The funny thing is that I've done it in the past for resizing or rotating and it works flawlessly. Passing a variable to the -font option is tripping up ImageMagick somehow.

Thanks again.

**Reply by fmw42** [reply]
This works fine for me. Note the quotes at the end.

```
FONT="/Library/fonts/candice.ttf"
convert -size 800x -gravity Center -background white -fill black \
-font "$FONT" -pointsize 72 \
caption:"This is Fred's Test" "$FONT.png"

```

Note that using an output as `"$FONT.png"` will make a file candice.ttf.png in the fonts directory.

**Reply by greytomorrow** [reply]
Fred, thanks for struggling through this with me. The final working result:

```
#! /bin/bash

read -p "Drag font here: " FONT
echo "$FONT"

convert -size 800x -gravity Center -background white -fill black \
-font "$FONT" -pointsize 72 \
caption:@/fakepath/00_alphabet.txt "$FONT.png"
```

**Comment by snibgo** [comment]
Your output filename also needs to be quoted.
